### ISSUE TITLE
Acid should dissolve objects. 

### ISSUE THREAD

**Issue by TotallyNotALier** [studio-minus/ppg-bugs-and-requests#637]
**Feature description**
Acid will be able to dissolve metalic, wooden and maybe some other objects (like weapons). Special shader aplies to objects damaged by acid, after some time they disappear. 

**Feature purpose**
More realism. Acid currently dissolves flesh, which is a bit unrealistic. Second, destroying metal objects is impossible, so acid will solve this problem. 

**Additional context**
No.

**Comment by DindinYT37** [comment]
If we get multiple levels of acid, like 1 (Some flesh burns/dissolving) to 5 (even dissolves metal), it would be nice too.

**Comment by 404CoderNotFound** [comment]
> If we get multiple levels of acid, like 1 (Some flesh burns/dissolving) to 5 (even dissolves metal), it would be nice too.

Good idea, but how would the dissolving work? It'd be way too hard to make something like per-pixel destruction without ruining performance. And also, how would we get the acid to touch the metal? With the shower head?

**Comment by DindinYT37** [comment]
> Good idea, but how would the dissolving work?

Well you could do per X pixel destruction, which could be adjustable in settings to control performance (maybe make default 4).

> And also, how would we get the acid to touch the metal? With the shower head?

Test tubes, kind of like flasks but intended for acid and can spill. Also, Flasks should break if Acid is inserted into them.

**Comment by DavidCo113** [comment]
> > Good idea, but how would the dissolving work?
> 
> Well you could do per X pixel destruction, which could be adjustable in settings to control performance (maybe make default 4).
> 
> > And also, how would we get the acid to touch the metal? With the shower head?
> 
> Test tubes, kind of like flasks but intended for acid and can spill. _Also, Flasks should break if Acid is inserted into them._

The flask is made of an indestructable divine substance called "polymethylmethacrylaat". Glass is already acid immune but this is even more immune.
In addition, a Gorse Blood flask exists. Acid is just a recolor of Gorse Blood, so I don't think flasks should shatter with acid.

**Comment by DindinYT37** [comment]
> > > Good idea, but how would the dissolving work?
> > 
> > 
> > Well you could do per X pixel destruction, which could be adjustable in settings to control performance (maybe make default 4).
> > > And also, how would we get the acid to touch the metal? With the shower head?
> > 
> > 
> > Test tubes, kind of like flasks but intended for acid and can spill. _Also, Flasks should break if Acid is inserted into them._
> 
> The flask is made of an indestructable divine substance called "polymethylmethacrylaat". Glass is already acid immune but this is even more immune. In addition, a Gorse Blood flask exists. Acid is just a recolor of Gorse Blood, so I don't think flasks should shatter with acid.

You see, Gorse Blood is Level 1/2 Acid, while only Level 3+ Acids shatter flasks.
Also, the game doesn't have to be 1:1 realistic, flasks don't shatter in the game in the first place which could also mean it's another translucent, acid-resistant material.

**Comment by 404CoderNotFound** [comment]
> > > Good idea, but how would the dissolving work?
> > 
> > 
> > Well you could do per X pixel destruction, which could be adjustable in settings to control performance (maybe make default 4).
> > > And also, how would we get the acid to touch the metal? With the shower head?
> > 
> > 
> > Test tubes, kind of like flasks but intended for acid and can spill. _Also, Flasks should break if Acid is inserted into them._
> 
> The flask is made of an indestructable divine substance called "polymethylmethacrylaat". Glass is already acid immune but this is even more immune. In addition, a Gorse Blood flask exists. Acid is just a recolor of Gorse Blood, so I don't think flasks should shatter with acid.

Polymethylmethacrylaat is dutch for Polymethylmethacrylate, also known as Plexiglass, which is in fact only resistant to really weak acids, but definitely not to acids that can dissolve human flesh. For example, Acetone can't dissolve a human body. However, you could completely dissolve plastic into acetone. This is why zooi just said "Alright, then i'll just say it's indestructible". He can still change that tho.

**Comment by 404CoderNotFound** [comment]
> > > > Good idea, but how would the dissolving work?
> > > 
> > > 
> > > Well you could do per X pixel destruction, which could be adjustable in settings to control performance (maybe make default 4).
> > > > And also, how would we get the acid to touch the metal? With the shower head?
> > > 
> > > 
> > > Test tubes, kind of like flasks but intended for acid and can spill. _Also, Flasks should break if Acid is inserted into them._
> > 
> > 
> > The flask is made of an indestructable divine substance called "polymethylmethacrylaat". Glass is already acid immune but this is even more immune. In addition, a Gorse Blood flask exists. Acid is just a recolor of Gorse Blood, so I don't think flasks should shatter with acid.
> 
> You see, Gorse Blood is Level 1/2 Acid, while only Level 3+ Acids shatter flasks. Also, the game doesn't have to be 1:1 realistic, flasks don't shatter in the game in the first place which could also mean it's another translucent, acid-resistant material.

Then we could even make it so that bones dissolve too in certain acids. But, that'd be probably even harder to code correctly when it comes to colliders and stuff.

**Comment by 404CoderNotFound** [comment]
Perhaps the whole acid thing could be added in a "chemistry update" with other chemicals?

**Comment by DindinYT37** [comment]
> Perhaps the whole acid thing could be added in a "chemistry update" with other chemicals?

You know, what we have talked about should be a new feature request since it has become quite large and distant from this request.

**Comment by DavidCo113** [comment]
> > The flask is made of an indestructable divine substance called "polymethylmethacrylaat". Glass is already acid immune but this is even more immune. In addition, a Gorse Blood flask exists. Acid is just a recolor of Gorse Blood, so I don't think flasks should shatter with acid.
> 
> Polymethylmethacrylaat is dutch for Polymethylmethacrylate, also known as Plexiglass, which is in fact only resistant to really weak acids, but definitely not to acids that can dissolve human flesh. For example, Acetone can't dissolve a human body. However, you could completely dissolve plastic into acetone. This is why zooi just said "Alright, then i'll just say it's indestructible". He can still change that tho.

Huh. Didn't know that.

**Comment by DavidCo113** [comment]
The name and being called a "divine substance" made me think it wasn't a real thing and I didn't bother looking it up.

**Comment by YourLocalPlonker** [comment]
> If we get multiple levels of acid, like 1 (Some flesh burns/dissolving) to 5 (even dissolves metal), it would be nice too.

Do you mean something like ph in real life? Seems good

**Comment by DindinYT37** [comment]
> > If we get multiple levels of acid, like 1 (Some flesh burns/dissolving) to 5 (even dissolves metal), it would be nice too.
> 
> Do you mean something like ph in real life? Seems good

Basically, yeah. But way more simplified.

**Comment by mestiez** [comment]
I love observing chemistry but I am uninterested in studying the subject so I know nothing about it. I'd like to add semi-proper fun chemistry to the game but I'm just too dumb.
